pysoa/common/logging.py

```diff
@@ -199,3 +202,179 @@ def _copy_and_censor_dict(cls, d):
     @classmethod
     def _copy_and_censor_iterable(cls, i, should_censor_strings):
         return type(i)(cls._copy_and_censor_unknown_value(v, should_censor_strings) for v in i)
+
+
+IP_MTU_DISCOVER = 10  # Position of the IP Path MTU Discovery flag in request packets
+IP_MTU_DISCOVER_DO = 2  # "Don't fragment" value of the IP Path MTU Discovery flag in request packets
+WORST_CASE_MTU_IP = 576
+MTU_ATTEMPTS = 65535, 2000, 1500, 1280
+DATAGRAM_HEADER_LENGTH_IN_BYTES = 28
+
+
+def _discover_minimum_mtu_to_target(address, port):
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    s.connect((address, port))
+    s.setsockopt(socket.IPPROTO_IP, IP_MTU_DISCOVER, IP_MTU_DISCOVER_DO)
+    for attempt in MTU_ATTEMPTS:
+        try:
+            s.send(b'#' * (attempt - DATAGRAM_HEADER_LENGTH_IN_BYTES))
+        except socket.error as e:
+            if 'too long' not in e.strerror:
+                # We can't rely on the error code, because it varies from platform to platform, but the message always
+                # contains "too long" if we're getting an "MTU exceeded" ICMP response. IF this isn't an "MTU exceeded"
+                # response, then something went wrong, and we can't determine the MTU, so fall back to worst case.
+                return WORST_CASE_MTU_IP
+            # Now we try again with a lower MTU, which might still be too high
+        else:
+            return attempt
+    return WORST_CASE_MTU_IP
+
+
+class SyslogHandler(logging.handlers.SysLogHandler):
```

Comment:
Please rename to `SysLogHandler` (not the capital `L`) to be consistent with `logging.handlers.SysLogHandler` and config issues... this is very likely to get mistyped otherwise

Reply:
I disagree with this. Point of fact: The handler name in Python is incorrect (just like all the time zone stuff in Python is misnamed and should be `time_zone`/`TimeZone` instead of `timezone`/`Timezone`). The proper name is Syslog, not SysLog, so the correct name for this handler is `SyslogHandler`, not `SysLogHandler`. I'd rather have a correct name that doesn't match Python than an incorrect name that does match Python.